content/blog/2018-10-23-react-v-16-6.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "React v16.6.0: lazy, memo and contextType"
+author: [sebmarkbage]
+---
+
+Today we're releasing React 16.6 with a few new convenient features. An easier way to consume Context from class components, a form of PureComponent/shouldComponentUpdate for function components and a way to do code splitting using Suspense.
+
+Check out the full [changelog](#changelog) below.
+
+## [`React.memo`](/docs/reference-react.md#reactmemo)
+
+Class components can bail out from rendering when their input props are the same using [`PureComponent`](/docs/react-api.html#reactpurecomponent) or [`shouldComponentUpdate`](/docs/react-component.html#shouldcomponentupdate). Now you can do the same with function components by wrapping them in [`React.memo`](/docs/reference-react.md#reactmemo).
+
+```js
+const MyComponent = React.memo(function MyComponent(props) {
+  /* only rerenders if props change */
+});
+```
+
+## [`React.lazy`](/docs/code-splitting.md#reactlazy) - Code-Splitting with `Suspense`
+
+You may have seen [Dan's talk about React Suspense at JSConf Iceland](/blog/2018/03/01/sneak-peek-beyond-react-16.html). Now you can use the Suspense component to do [code-splitting](/docs/code-splitting.md#reactlazy) by wrapping a dynamic import in a call to `React.lazy()`.
+
+```js
+import React, {lazy, Suspense} from 'react';
+const OtherComponent = lazy(() => import('./OtherComponent'));
+const MyComponent = () => (
```

Comment:
Mind using function declaration? We use it pretty much everywhere in docs.

Reply:
We're very inconsistent but sure.